### DISCUSSION TITLE
Discuss how to remove blocks on a granular level from celestia-node

### DISCUSSION THREAD

**Discussion by renaynay** [celestiaorg/celestia-node#3119] [General]
Needs better description and title

**Comment by Wondertan** [comment]
Noting that here that the original motivation behind this can be also resolved by introducing CoreGetter for EDSes

**Comment by renaynay** [comment]
This (CoreGetter) *only* automatically resolves the issue in the case that we miss a block / do not store its data for some reason. It does not resolve the case in which we need to roll back some part of the chain for whatever reason (testing or otherwise) where technically valid blocks were already stored.
